testinit/testing-server/test_server.py

```diff
@@ -0,0 +1,137 @@
+import subprocess
+import threading
+import time
+import signal
+import sys
+from http.server import HTTPServer
+
+from config import WIRESERVER_IP, WIRESERVER_PORT, IMDS_IP, IMDS_PORT, DUMMY_IFACE
+from utils import logger, run_cmd
+
+from wireserver_handler import WireServerHandler
+from imds_handler import IMDSHandler
+
+class TestServer:
+    """Main test server class that manages network setup and HTTP servers."""
+    
+    def __init__(self):
+        self.imds_server = None
+        self.wireserver_server = None
+        self.imds_thread = None
+        self.wireserver_thread = None
+        self.running = False
+    
+    def setup_network_interface(self):
+        """Set up dummy network interface with required IP addresses."""
+        logger.info("Setting up dummy network interface...")
+        
+        try:
+            # Create dummy interface
+            run_cmd(f"sudo ip link add {DUMMY_IFACE} type dummy")
+            run_cmd(f"sudo ip link set {DUMMY_IFACE} up")
+            
+            # Assign IP addresses
+            run_cmd(f"sudo ip addr add {IMDS_IP}/32 dev {DUMMY_IFACE}")
+            run_cmd(f"sudo ip addr add {WIRESERVER_IP}/32 dev {DUMMY_IFACE}")
+            
+            # Add routes to ensure traffic goes to our dummy interface
+            run_cmd(f"sudo ip route add {IMDS_IP} dev {DUMMY_IFACE}")
+            run_cmd(f"sudo ip route add {WIRESERVER_IP} dev {DUMMY_IFACE}")
+            
+            logger.info("Network interface setup completed successfully")
+            
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Failed to set up network interface: {e}")
+            raise
+    
+    def cleanup_network_interface(self):
+        """Clean up dummy network interface."""
+        logger.info("Cleaning up network interface...")
+        
+        try:
+            # Remove routes first
+            run_cmd(f"sudo ip route del {IMDS_IP} dev {DUMMY_IFACE} 2>/dev/null || true")
+            run_cmd(f"sudo ip route del {WIRESERVER_IP} dev {DUMMY_IFACE} 2>/dev/null || true")
+            
+            # Remove the dummy interface
+            run_cmd(f"sudo ip link delete {DUMMY_IFACE} 2>/dev/null || true")
+            
+            logger.info("Network interface cleanup completed")
+            
+        except subprocess.CalledProcessError as e:
+            logger.warning(f"Error during cleanup (this might be expected): {e}")
+    
+    def start_imds_server(self):
```

Comment:
At some point it would be nice to feed a "recipe/playbook" to the mock IMDS server (via a file passed to command line) and ask it to do stuff like
"Respond with 500 for the first request (test http 500 handling), sleep on the next request (test read timeout handling), respond with 404, respond with partial data, respond with full data"
 
In normal mode you can make the server keep responding with good data, but in a "playbook" mode you can use it to test azure-init handling of various IMDS/wireserver condition